tachyon/crypto/commitments/polynomial_openings_unittest.cc

```diff
@@ -118,10 +123,16 @@ TEST_F(PolynomialOpeningsTest, GroupByPolyOracleAndPoints) {
                              polys_[1].Evaluate(points_[1]));
   poly_openings.emplace_back(PolyRef(&polys_[2]), PointDeepRef(&points_[2]),
                              polys_[2].Evaluate(points_[2]));
+  poly_openings.emplace_back(PolyRef(&polys_[3]), PointDeepRef(&points_[2]),
+                             polys_[3].Evaluate(points_[2]));
   PolynomialOpeningGrouper<Poly> grouper;
   std::vector<GroupedPolyOraclePair> poly_openings_grouped_by_poly =
       grouper.GroupByPolyOracle(poly_openings);
 
+  // Although |polys_[2]| and |polys_[3]| have the same value, they should be
+  // grouped separately because they are ShallowRef in |PolynomialOpenings|.
```

Comment:
How about adding `Note(Insun35):` here as well?

Reply:
NOTE should be all upper cases! `NOTE(Insun35)`